### ISSUE TITLE
events.xml event with exclude param

### ISSUE THREAD

**Issue by Michal-sk** [magento/magento2#1684]
I would like to hook on the event ```core_collection_abstract_load_before```. But in my observer i would like to inject the ```\Magento\Backend\Model\Auth```, so i can get the logged in admin. This gives problems because Somewhere in the injecting process of Auth, we try to inject the ```StoreManager``` which itself again tries to load a collection of stores. This breaks the whole process and means i will have to manually define all the events that use ```load_before```, but than with the ```event_prefix```.

Defining an except in the ```<event>``` tag would result in less work. We could even make it accept a list?
```xml
    <event name="core_collection_abstract_load_before" exclude="store_collection_load_before">
        <observer name="foo_bar_event_prefixed_collection_load_before" instance="Foo\Bar\Model\Observer" method="coreCollectionLoadBefore" />
    </event>
```

Or is maybe something already possible?

**Comment by Michal-sk** [comment]
This comes down to not being able to use ```core_collection_abstract_load_before``` and injecting a session related class. Keep get getting ```Store Manager has been initialized not properly```. What could be another way to check in my observer if a admin is loggedin and get the admin user.

**Comment by Michal-sk** [comment]
Ok. So i wanted to observer ```core_collection_abstract_load_before``` and inject ```\Magento\Backend\Model\Auth``` in my observer. But since we listen to a very generic event, this become problematic when magento was loading websites, groups, stores and config (magento loads config to get session related settings).

This is how i overcome this issue and am able to use the observer as generic as posible:
```php
<?php
namespace Ms\UserWebsite\Model;

class InvokerPlugin
{
    public function beforeDispatch(
        \Magento\Framework\Event\Invoker\InvokerDefault $subject,
        array $configuration,
        \Magento\Framework\Event\Observer $observer
    ) {
        if (isset($configuration['instance']) && $configuration['instance'] === 'Ms\UserWebsite\Model\Observer') {
            if ($observer->getCollection() instanceof \Magento\Store\Model\Resource\Website\Collection
                || $observer->getCollection() instanceof \Magento\Store\Model\Resource\Group\Collection
                || $observer->getCollection() instanceof \Magento\Store\Model\Resource\Store\Collection
                || $observer->getCollection() instanceof \Magento\Store\Model\Resource\Config\Collection\Scoped
            ) {
                $configuration['disabled'] = true;
                return [$configuration, $observer];
            }
        }

        return [$configuration, $observer];
    }
}
```

Excluding the store related collection because of ```Store Manager has been initialized not properly``` and the config because of a ```Circular dependency LogicException```. My question remains tough. What would be a good way to handle this situation. Why i dont want to call all the observer with there ```event_name``` prefixed is because not all collections have a ```event_name``` which would not call my observer. The way i handled it now has a hacky taste to it, tough working nice. We could ofcourse solve this with excluding events, tough that would give the same issue, what if you want to exclude an event that does not have a ```event_name```.

**Comment by Michal-sk** [comment]
Gues i wasn't the only one with this problem; ```'Magento\Store\Model\Resource\Website\Collection::getLoadDefault``` to the rescue.

**Comment by magento-engcom-team** [comment]
Hi @Michal-sk. Thank you for your report. 
The issue has been fixed in magento/magento2#29398 by @yolouiese in 2.4-develop branch
Related commit(s):
 - [deb001c63d195c7c0d5d0cebc0fbb38bc622d58b](https://github.com/magento/magento2/commit/deb001c63d195c7c0d5d0cebc0fbb38bc622d58b)
 - [b5550cc239f57bcf3dc5dd8279efac9da40dcff6](https://github.com/magento/magento2/commit/b5550cc239f57bcf3dc5dd8279efac9da40dcff6)
 - [cc941330b74cad34ce3200c8e8b092d10efd6eb3](https://github.com/magento/magento2/commit/cc941330b74cad34ce3200c8e8b092d10efd6eb3)
 - [2d0bbadf541d0465bee76f1ca26e0c82ec23fb14](https://github.com/magento/magento2/commit/2d0bbadf541d0465bee76f1ca26e0c82ec23fb14)
 - [811c4ad9f6082d464de8bef410f2b31eda9d4a4e](https://github.com/magento/magento2/commit/811c4ad9f6082d464de8bef410f2b31eda9d4a4e)
 - [88923fa23b33b86a1ad483fe962a8f405616a692](https://github.com/magento/magento2/commit/88923fa23b33b86a1ad483fe962a8f405616a692)
 - [07dd8956b7b49782b8bb112bb48f67f1e76a576c](https://github.com/magento/magento2/commit/07dd8956b7b49782b8bb112bb48f67f1e76a576c)
 - [ad3da2d131a877c81d86875cbb3bcafe41479fce](https://github.com/magento/magento2/commit/ad3da2d131a877c81d86875cbb3bcafe41479fce)

The fix will be available with the upcoming 2.4.1 release.
